lib/events.dart

```diff
@@ -1,38 +1,130 @@
 part of pdftron;
 
-const _exportAnnotationCommandChannel = const EventChannel('export_annotation_command_event');
+const _exportAnnotationCommandChannel =
+    const EventChannel('export_annotation_command_event');
 const _exportBookmarkChannel = const EventChannel('export_bookmark_event');
 const _documentLoadedChannel = const EventChannel('document_loaded_event');
+const _documentErrorChannel = const EventChannel('document_error_event');
+const _annotationChangedChannel =
+    const EventChannel('annotation_changed_event');
+const _annotationsSelectedChannel =
+    const EventChannel('annotations_selected_event');
+const _formFieldValueChangedChannel =
+    const EventChannel('form_field_value_changed_event');
 
 typedef void ExportAnnotationCommandListener(dynamic xfdfCommand);
 typedef void ExportBookmarkListener(dynamic bookmarkJson);
 typedef void DocumentLoadedListener(dynamic filePath);
+typedef void DocumentErrorListener();
+typedef void AnnotationChangedListener(dynamic action, dynamic annotations);
+typedef void AnnotationsSelectedListener(dynamic annotationWithRects);
+typedef void FormFieldValueChangedListener(dynamic fields);
 typedef void CancelListener();
 
 const int exportAnnotationId = 1;
 const int exportBookmarkId = 2;
 const int documentLoadedId = 3;
+const int documentErrorId = 4;
+const int annotationChangedId = 5;
+const int annotationsSelectedId = 6;
+const int formFieldValueChangedId = 7;
```

Comment:
As above, would be preferable to use an enum here. Since these values are shared, would be better if they could be declared once in a format that both dart and Obj-C can both use. I am probably dreaming, but mentioning just in case it's possible!